lib/CleantalkSP/Templates/Singleton.php

```diff
@@ -18,12 +18,12 @@ public function __clone(){}
 	 * @return $this
 	 */
 	public static function getInstance($params = array()){
-		if (!isset(static::$instance)) {
-			static::$instance = new static;
+		if( ! isset( static::$instance[ static::class ] ) ){
+			static::$instance[ static::class ] = new static;
             $params = array_values($params);
-			static::$instance->init(...$params);
+			static::$instance[ static::class ]->init(...$params);

```

Comment:
I am against of variable-length arguments

Reply:
Noticed. We'll discuss it.